### PR TITLE
Release using github actions

### LINKED ISSUE #45: Move k8s-integration-release Job
https://fsi-build.pdx.vm.datanerd.us/job/k8s-integration-release/

Currently it downloads an image from a Private Quay repo, that image was the one built and tested, however there is no reason why we should not rebuild it.

It tags it, and uploads it to docker hub.

It also uploads release manifest to S3

The expected behaviour is:
Running tests on PR, merge to master/main
Created a release when creating a new tag in the repo

### PR DESCRIPTION
**Pull request by roobre**
Depends on #49 
Closes #45

- [x] Release pipeline structure
- [x] Rebase #49

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      - uses: actions/checkout@v2
+
+      - name: Build docker images
+        run: |
+          make compile
+          docker build -t $DOCKER_IMAGE_NAME:$DOCKER_IMAGE_TAG .
+          docker build -t $DOCKER_IMAGE_NAME:$DOCKER_IMAGE_TAG-unprivileged --build-arg 'MODE=unprivileged' .
+#          DOCKER_PRIVILEGED_SHA256=$(docker inspect --format='{{index (split (index .RepoDigests 0) "@sha256:") 1}}' $DOCKER_IMAGE_NAME:$DOCKER_IMAGE_TAG)
+#          DOCKER_UNPRIVILEGED_SHA256=$(docker inspect --format='{{index (split (index .RepoDigests 0) "@sha256:") 1}}' $DOCKER_IMAGE_NAME:$DOCKER_IMAGE_TAG-unprivileged)
+#          test "$DOCKER_PRIVILEGED_SHA256" != "$DOCKER_UNPRIVILEGED_SHA256"
```
Comment (roobre) [inline .github/workflows/release.yml:36]: This check was present on Jenkins, although it cannot be directly ported here as locally generated images have an empty `.RepoDigests`. I can rewrite this to extract the SHA from the image ID if you think it is worth it, although I do not really see much value on it.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+          docker push $DOCKER_IMAGE_NAME:$DOCKER_IMAGE_TAG
+          docker push $DOCKER_IMAGE_NAME:$DOCKER_IMAGE_TAG-unprivileged
+
+#      - name: Tag and push docker :latest images
+#        if: ${{ ! github.event.release.prerelease }}
+#        run: |
+#          docker tag $DOCKER_IMAGE_NAME:$DOCKER_IMAGE_TAG $DOCKER_IMAGE_NAME:latest
+#          docker tag $DOCKER_IMAGE_NAME:$DOCKER_IMAGE_TAG-unprivileged $DOCKER_IMAGE_NAME:latest-unprivileged
+#          docker push $DOCKER_IMAGE_NAME:latest
+#          docker push $DOCKER_IMAGE_NAME:latest-unprivileged
```
Comment (roobre) [inline .github/workflows/release.yml:54]: Jenkins release pipeline does not push `:latest` images. I don't know the rationale for this decision, and in all other integrations we are pushing `:latest`, what are your thoughts on this?

Comment (roobre) [reply .github/workflows/release.yml:54]: We agreed to push latest to keep consistency between all release pipelines.

File: .github/workflows/release.yml (hunk 1 above)
Comment (roobre) [reply .github/workflows/release.yml]: We agreed this was not very useful and stripped it completely

**Review by arvdias — DISMISSED** [review]

**Review by gsanchezgavier — DISMISSED** [review]

**Comment by roobre** [comment]
Rebased #49

**Review by gsanchezgavier — APPROVED** [review]

**Review by paologallinaharbur — APPROVED** [review]

**Pull request merged**